### ISSUE TITLE
[Bug] Cannot load settings; use safe access in 2 places

### ISSUE THREAD

**Issue by kometenstaub** [Fevol/obsidian-translate#22]
```
plugin:obsidian-translate:42349 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'default_usage')
    at setupLanguageDetector (plugin:obsidian-translate:42349:47)
    at setupTranslationService (plugin:obsidian-translate:42365:5)
    at Object.$$self.$$.update (plugin:obsidian-translate:42596:27)
    at init (plugin:obsidian-translate:1248:6)
    at new Reactivity (plugin:obsidian-translate:42646:5)
    at TranslatorPlugin.onload (plugin:obsidian-translate:42765:23)
```

![screenshot](https://user-images.githubusercontent.com/83140328/180667906-1c6c69e0-8f43-4384-b648-cb73a434168e.png)

The plugin cannot load the settings properly. The commands don't get loaded properly and after clicking on the `Translate` Tab in the community plugins, it is empty, and every new settings tab is empty as well and throws this error:

```
plugin:obsidian-translate:42699 Uncaught TypeError: Cannot read properties of undefined (reading '$destroy')
    at TranslatorSettingsTab.hide (plugin:obsidian-translate:42699:15)
    at t.openTab (app.js:1:2063355)
    at HTMLDivElement.<anonymous> (app.js:1:2062226)
```

![screenshot](https://user-images.githubusercontent.com/83140328/180667940-5c55c29e-69da-483a-88bd-aa3e2ade8892.png)

Error when clicking on the translate tab:

```
plugin:obsidian-translate:27282 Uncaught TypeError: Cannot read properties of undefined (reading 'default_usage')
    at Array.create_control_slot (plugin:obsidian-translate:27282:47)
    at create_slot (plugin:obsidian-translate:515:29)
    at create_fragment9 (plugin:obsidian-translate:2809:28)
    at init (plugin:obsidian-translate:1251:41)
    at new SettingItem (plugin:obsidian-translate:2927:9)
    at create_fragment15 (plugin:obsidian-translate:27406:18)
    at init (plugin:obsidian-translate:1251:41)
    at new SettingsPage (plugin:obsidian-translate:28094:5)
    at TranslatorSettingsTab.display (plugin:obsidian-translate:42687:17)
    at t.openTab (app.js:1:2063766)
```

![screenshot](https://user-images.githubusercontent.com/83140328/180668054-4a467656-e2ab-48dc-bbfa-6e128b4c92ca.png)

With a safe access (`?`) in these two places, it works again.

**Comment by kometenstaub** [comment]
I don't know why it happens because it is in the default settings. :thinking:

**Comment by Fevol** [comment]
Ah, that'll be because the `data.json` of older versions does not automatically get updated with new default settings. Big oversight on my part!

As for a fix, I'm thinking the best solution would be to just add any missing values at plugin start-up, so it won't be necessary to have a bunch of safe navigation operators everywhere. There are quite a few things about the way the settings are handled right now that I'm unhappy about, I'll try to release a fix asap (today hopefully)!

**Comment by Fevol** [comment]
Sorry for the wait! I'll be shipping the fix for this issue tomorrow (or the day after that), together with the new translation services.

I've overhauled the settings handler so it always keeps the data.json in sync with the default settings (though this comes at the cost of an additional 20ms of start-up time, not exactly sure if I'll be able to reduce that). But in general, this error shouldn't be able to happen anymore

**Comment by Fevol** [comment]
Well, it took two weeks, but I finally added the fix to release [1.3.0](https://github.com/Fevol/obsidian-translate/releases/tag/1.3.0), hopefully it works for you!

**Comment by kometenstaub** [comment]
Yes, it works now. Thank you!
